lib/shoryuken/default_worker_registry.rb

```diff
@@ -18,6 +18,11 @@ def fetch_worker(queue, message)
         message.message_attributes['shoryuken_class'] &&
         message.message_attributes['shoryuken_class'][:string_value]
 
+      approximate_receive_count = message.attributes['ApproximateReceiveCount'].to_i
+      if approximate_receive_count > 1 && 'ActiveJob::QueueAdapters::ShoryukenAdapter::JobWrapper' != worker_class
```

Comment:
This code needs to disappear, but is was a hacky way add support for Shoryuken workers as wel.

When you are using the worker class implementation, the worker should be fetched based on the queue name. (@workers[queue])
In case of redelivery, you want to ignore the message attributes, and clear the worker_class if it's already set so @workers[queue] is executed.

When using a Shoryuken worker the implementation can be really clean, just @workers[queue].new

The AJ adapter is complicating things and making this method complex.